components/expression_language/ast.rst

```diff
@@ -0,0 +1,33 @@
+.. index::
+    single: AST; ExpressionLanguage
+
+Dumping and Manipulating the AST of Expressions
+===============================================
+
+In computer science, `AST`_ (*Abstract Syntax Trees*) are a tree representation
+of the structure of source code written in a programming language. The
+expressions created with the ExpressionLanguage component are strings, which
+make them difficult to manipulate or inspect.
+
+A better approach is to dump the AST of those expressions using the ``dump()``
+method. This turns the original string expression into a set of PHP classes
+describing the operations of that expression::
+
+    use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+
+    $language = new ExpressionLanguage();
+    $ast = $language->dump('1 + 2');
```

Comment:
Hi;

Actually, this method does not exist. You can get the AST like this:

```php
<?php

require __DIR__.'/vendor/autoload.php';

use Symfony\Component\ExpressionLanguage\ExpressionLanguage;

$ast = (new ExpressionLanguage())
        ->parse('var.a.b + 12', ['var'])
        ->getNodes()
;

dump($ast);
```

And then you can dump the AST (as a string) `dump($ast->dump());`

Finally, Instead of getting a string, it's possible to get an array in order to manipulate it: `dump($ast->toArray());`